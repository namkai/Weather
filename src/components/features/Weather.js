import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import CurrentWeather from '../common/CurrentWeather';
import WeatherHeader from '../common/WeatherHeader';
import selectedForecast from '../../selectors/selected_forecast';
import Chart from '../common/Chart';

export const Weather = ({ weather, forecast }) => (
  <div className="col-md-9 content">
    <WeatherHeader city={weather.name} description={weather.weather}/>
    <hr className="mt-3"/>
    <CurrentWeather {...weather} />
    <Chart forecast={forecast} />
  </div>
);

Weather.defaultProps = {
  weather: {},
  forecast: {
    list: [],
  },
};

Weather.propTypes = {
  weather: PropTypes.object,
  forecast: PropTypes.arrayOf(PropTypes.array),
};

const mapStateToProps = state => ({
  weather: state.weather,
  forecast: selectedForecast(state),
});


export default connect(mapStateToProps)(Weather);
