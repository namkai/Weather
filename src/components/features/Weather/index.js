import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import CurrentWeather from '../../common/CurrentWeather/index';
import WeatherHeader from '../../common/WeatherHeader/view';
import Chart from '../../common/Chart/index';
import WeatherLayout from '../../layouts/WeatherLayout';

export const Weather = ({ weather, forecast }) => (
  <WeatherLayout header={<WeatherHeader city={weather.name} description={weather.weather} />} >
    <CurrentWeather {...weather} />
    <Chart forecast={forecast} />
  </WeatherLayout>
);

Weather.defaultProps = {
  weather: {},
  forecast: {
    list: [],
  },
};

Weather.propTypes = {
  weather: PropTypes.object,
  forecast: PropTypes.arrayOf(PropTypes.array),
};

export default Weather;
