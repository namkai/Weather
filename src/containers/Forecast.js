import 'c3/c3.css';
import React from 'react';
import { connect } from 'react-redux';
import selectedForecast from '../selectors/selected_forecast';
import Chart from '../components/Chart';

const Forecast = ({ forecast }) => {
	console.log(forecast, 'i\'m the forecast');
	if (forecast.length === 0) {
		return (
			<div></div>
		);

	}
	const data = {
		columns: [
			['Temperature', ...forecast],
		],
		axis: {
			type: 'timeseries',
			tick: {
				format: '%Y-%m-%d',
			},
		},
		color: 'orange',
	};
	return (
		<Chart forecast={forecast} chartType={'bar'} />
	);
};

const mapStateToProps = (state) => ({ forecast: selectedForecast(state) });

export default connect(mapStateToProps)(Forecast);
