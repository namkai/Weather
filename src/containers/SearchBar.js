import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/index';

export const SearchBar = ({ input, updateQuery, fetchWeather }) => {
	const searchForWeather = (e) => {
		e.preventDefault();
		fetchWeather(input);
	};
	return (
    <form onSubmit={searchForWeather} className="input-group">
      <input
        placeholder="Get a five-day forecast in your favorite cities"
        className="form-control"
        value={input}
        onChange={e => updateQuery(e.target.value)}
      />
      <span className="input-group-btn">
        <button type="submit" className="btn btn-secondary">Submit</button>
      </span>
    </form>
	);
};

SearchBar.defaultProps = {
	input: '',
	updateQuery: PropTypes.func,
	fetchWeather: PropTypes.func,
};

SearchBar.propTypes = {
	input: PropTypes.string,
	updateQuery: PropTypes.func,
	fetchWeather: PropTypes.func,
};

export default connect(({ input }) => ({ input }), actions)(SearchBar);
