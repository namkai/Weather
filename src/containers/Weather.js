import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import CurrentWeather from '../components/CurrentWeather';
import WeatherHeader from '../components/WeatherHeader';

export const Weather = ({ weather }) => {
	return (
  <div className="col-md-9 content">
    <WeatherHeader city={weather.name} description={weather.weather} />
    <hr className="mt-3" />
    <CurrentWeather {...weather} />
  </div>
	);
};

Weather.defaultProps = {
	weather: {},
};

Weather.propTypes = {
	weather: PropTypes.object,
};

export default connect(({ weather, forecast }) => ({ weather, forecast }))(Weather);
