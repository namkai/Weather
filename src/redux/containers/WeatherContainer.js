import { compose, pure } from 'recompose';
import { connect } from 'react-redux';

import getWeatherState from '../selectors/getWeatherState';
import Weather from '../../components/features/Weather';

function mapStateToProps(state) {
  return getWeatherState(state);
}

const connectToStore = connect(mapStateToProps);

const PureWeather = pure(Weather);

export default compose(connectToStore)(PureWeather);
