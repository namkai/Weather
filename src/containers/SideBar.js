import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import Cities from '../components/Cities';
import SearchBar from '../components/SearchBar';

export const SideBar = ({ storedCities, fetchWeather, updateQuery, input }) => {
	return (
		<div className="col-md-3 sidebar">
			<nav className="sidebar-nav">
				<div className="sidebar-header">
					<button className="nav-toggler nav-toggler-md sidebar-toggler" type="button" data-toggle="collapse"
					        data-target="#nav-toggleable-md">
						<span className="sr-only">Toggle nav</span>
					</button>
					<a className="sidebar-brand img-responsive" href="#">
						<span style={{ color: '#1ec8ff' }} className="icon icon-drop sidebar-brand-icon"/>
					</a>
				</div>

				<div className="collapse nav-toggleable-md" id="nav-toggleable-md">
					<SearchBar updateQuery={updateQuery} fetchWeather={fetchWeather} input={input}/>
					<Cities storedCities={storedCities} fetchWeather={fetchWeather}/>
					<hr className="visible-xs mt-3"/>
				</div>
			</nav>
		</div>
	);
};

SideBar.defaultProps = {
	storedCities: PropTypes.arrayOf(PropTypes.object),
	fetchWeather: PropTypes.func,
	updateQuery: PropTypes.func,
	input: PropTypes.string,
};

SideBar.propTypes = {
	storedCities: PropTypes.arrayOf(PropTypes.object),
	fetchWeather: PropTypes.func,
	updateQuery: PropTypes.func,
	input: PropTypes.string,
};

export default connect(({ storedCities, input }) => ({ storedCities, input }), actions)(SideBar);