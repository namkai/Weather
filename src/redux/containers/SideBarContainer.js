import { compose, lifecycle } from 'recompose';
import { connect } from 'react-redux';

import * as actions from '../actions'
import getSideBarState from '../selectors/getSideBarState';
import SideBar from '../../components/features/SideBar';

const mapStateToProps = state => getSideBarState(state);

const connectToStore = connect(mapStateToProps, actions);


export default compose(connectToStore)(SideBar);
