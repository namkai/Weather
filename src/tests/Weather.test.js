import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Weather } from '../containers/Weather';

it('renders correctly', () => {
	const weather = shallow(<Weather />);
	expect(toJson(weather)).toMatchSnapshot();
});
