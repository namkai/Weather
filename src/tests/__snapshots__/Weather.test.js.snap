// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col-md-9 content"
>
  <WeatherHeader
    city="Weather"
    description="Get the weather in any city"
  />
  <hr
    className="mt-3"
  />
  <CurrentWeather />
</div>
`;
